go/ros/ros2db3_to_mcap.go

```diff
@@ -168,27 +168,58 @@ type topicsRecord struct {
 	name                string
 	typ                 string
 	serializationFormat string
-	offeredQOSProfiles  string
+	offeredQOSProfiles  *string
+}
+
+func checkHasQOSProfiles(db *sql.DB) (bool, error) {
+	var count int
+	err := db.QueryRow(
+		`select count(*) from pragma_table_info('topics') where name = 'offered_qos_profiles'`,
+	).Scan(&count)
+	if err != nil {
+		return false, err
+	}
+	return count > 0, nil
 }
 
 func getTopics(db *sql.DB) ([]topicsRecord, error) {
-	rows, err := db.Query(
-		`select id, name, type, serialization_format, offered_qos_profiles from topics`,
-	)
+	hasQOSProfiles, err := checkHasQOSProfiles(db)
+	if err != nil {
+		return nil, err
+	}
+	var rows *sql.Rows
+	if hasQOSProfiles {
+		rows, err = db.Query(
+			`select id, name, type, serialization_format, offered_qos_profiles from topics`,
+		)
+	} else {
+		rows, err = db.Query(
+			`select id, name, type, serialization_format from topics`,
```

Comment:
can't you just `select *`?

Reply:
`select *` brings its own hassles. You need to land the values onto proper types, and the database library can't help you with that if you use `select *`, so you need to manually check for the existence of the columns you want, and verify they are the types you are expecting. Take a look at https://github.com/foxglove/mcap/pull/570/commits/b40cbc12935ccad81afd942951193cf23ef24f81 and see if you prefer it. After writing that I'm probably inclined to back it out and go back to two queries.

Reply:
Ok up to you, I haven't done go database stuff in a long time. If parsing the record is more work with `select *` then just having two queries seems fine.